db/query.go

```diff
@@ -50,6 +53,15 @@ func (q *Query) Reverse() *Query {
 	return q
 }
 
+// Offset causes the query to skip offset models when iterating through models
+// that match the query. Note that queries which use an offset have a runtime of
+// O(offset + N), where N is the number of models that match the query. This is
```

Comment:
Shouldn't it be `O(N - offset)`? Since you don't process the offset records?

Reply:
It depends on what you mean by "process". The key thing that I am trying to communicate here is that we have to actually iterate over `offset` keys before we do anything else. You can't just skip over `offset` keys for free. Imagine you have a linked list and need to find the nth item in the list. That's sort of what we need to do here.

So if `offset` is high, then that will be the dominating factor in the runtime. You could have a query that returns nothing but if `offset` is exceedingly high it will still take a long time to run. On the other hand, if `offset` is low and `N` is high, then `N` will be the dominating factor. 
One thing I technically didn't account for in this notation is the fact that `offset` won't have any additional affect on runtime if it is higher than the total number of keys in the index. But it might be more confusing to try to express that.

I suppose I should also clarify what I mean by "the number of models that match the query". I intended for this number to account for other parts of the query like `Max`. So if you set `Max` to 5, the number of models that match the query cannot exceed 5. I could change the wording a bit to make this more clear.

How do you feel about this explanation instead?

```
// Note that queries which use an offset have a runtime of O(max(K, offset) + N), where N is
// the number of models returned by the query and K is the total number of keys in the index.
// Queries with a high offset can take a long time to run, regardless of the number of
// models returned. This is due to limitations of the underlying database.
```

Reply:
We might also describe this as `O(offset + K + N)` since typically in big-O notation `O(max(a, b)) == O(a + b)` but I also think that might be confusing.

Reply:
Thanks for elaborating. That makes a lot more sense now. I prefer your new proposed comment. Let's also add a line about how this implements the same behavior as SQL's OFFSET, and link to it (you put it in the PR description, but think having this in the code is better for posterity's sake).